docs/user-guide/data-structures/creating-arrays-and-datasets.ipynb

```diff
@@ -0,0 +1,527 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "sc.vectors(dims=['x'], values=[[1, 2, 3], [4, 5, 6]])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Data Arrays\n",
+    "\n",
+    "There is essentially only one way to construct [data arrays](../../generated/classes/scipp.DataArray.rst), namely its initializer:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "x = sc.linspace('x', start=1.5, stop=3.0, num=4, unit='m')\n",
+    "a = sc.scalar('an attribute')\n",
+    "m = sc.array(dims=['x'], values=[True, False, True, False])\n",
+    "data = x ** 2\n",
+    "sc.DataArray(data, coords={'x': x}, attrs={'a': a}, masks={'m': m})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`coords`, `attrs`, and `masks` are optional but the `data` must always be given.\n",
+    "Note how the creation functions for `scipp.Variable` can be used to make the individual pieces of a data array.\n",
+    "\n",
+    "## Dataset\n",
+    "\n",
+    "[Datasets](../../generated/classes/scipp.Dataset.rst) are constructed by combining multiple data arrays or variables.\n",
+    "For instance, using the previously defined variables:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "sc.Dataset({'data1': data, 'data2': -data}, coords={'x': x})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Or from data arrays:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "da1 = sc.DataArray(data, coords={'x': x}, attrs={'a': a}, masks={'m': m})\n",
+    "da2 = sc.DataArray(-data, coords={'x': x})\n",
+    "sc.Dataset({'data1': da1, 'data2': da2})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Any Data Structure\n",
+    "\n",
+    "Any of `scipp.Variable`, `scipp.DataArray`, and `scipp.Dataset` and be created using the methods described in the following subsections.\n",
+    "\n",
+    "### From Files\n",
+    "\n",
+    "Scipp has a custom file format based on HDF5 which can store data structures.\n",
+    "See [Reading and Writing Files](../reading-and-writing-files.rst) for details.\n",
+    "In short, `scipp.io.open_hdf5` loads whatever Scipp object is stored in a given file.\n",
+    "For demonstration purposes, we use a `BytesIO` object here. But the same code can be used by passing a string as a file name to `to_hdf5` and `open_hdf5`."
```

Comment:
Also refer/link to scippnexus?

Reply:
I'd rather not. I think this is too application specific for these docs.

Reply:
Isn't it *more* likely that the reader has a Nexus file than a scipp-hdf5 file?

Reply:
Someone working at ESS, yes. But don't we want to keep the docs more general?

Reply:
Nexus is used in many places, including photon sources. Scipp is used only at ESS.

Reply:
But we want to keep scipp general and not link it to neutron scattering

Reply:
Nexus is *not* neutron-scattering specific.

Reply:
We are even linking to `scippneutron` and `ess` in the toolbar from *every* page in the docs, I really don't see what is wrong with mentioning `scippnexus` here, which has much wider applications.